### ISSUE TITLE
Fails while installing hpricot

### ISSUE THREAD

**Issue by viktorianer** [slim-template/html2slim#47]
What can I do? No idea -_-

```bash
$ bundle install

Installing hpricot 0.8.6 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/hpricot-0.8.6/ext/fast_xs
~/.asdf/installs/ruby/3.3.0/bin/ruby extconf.rb
checking for stdio.h... yes
creating Makefile

current directory: ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/hpricot-0.8.6/ext/fast_xs
make DESTDIR\= sitearchdir\=./.gem.20240322-49101-wmq9bj sitelibdir\=./.gem.20240322-49101-wmq9bj clean

current directory: ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/hpricot-0.8.6/ext/fast_xs
make DESTDIR\= sitearchdir\=./.gem.20240322-49101-wmq9bj sitelibdir\=./.gem.20240322-49101-wmq9bj
compiling fast_xs.c
fast_xs.c:165:39: error: incompatible function pointer types passing 'VALUE (VALUE)' (aka 'unsigned long (unsigned long)') to parameter of type 'VALUE (*)(VALUE, VALUE)' (aka 'unsigned long (*)(unsigned long,
unsigned long)') [-Wincompatible-function-pointer-types]
        array = rb_rescue(unpack_utf8, self, unpack_uchar, self);
                                             ^~~~~~~~~~~~
~/.asdf/installs/ruby/3.3.0/include/ruby-3.3.0/ruby/internal/iterator.h:364:62: note: passing argument to parameter 'r_proc' here
VALUE rb_rescue(VALUE (*b_proc)(VALUE), VALUE data1, VALUE (*r_proc)(VALUE, VALUE), VALUE data2);
                                                             ^
1 error generated.
make: *** [fast_xs.o] Error 1

make failed, exit code 2

Gem files will remain installed in ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/hpricot-0.8.6 for inspection.
Results logged to ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/extensions/arm64-darwin-23/3.3.0/hpricot-0.8.6/gem_make.out

  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:125:in `run'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:51:in `block in make'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:43:in `each'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:43:in `make'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/ext_conf_builder.rb:42:in `build'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:193:in `build_extension'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:227:in `block in build_extensions'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in `each'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in `build_extensions'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/3.3.0/rubygems/installer.rb:852:in `build_extensions'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/rubygems_gem_installer.rb:72:in `build_extensions'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/rubygems_gem_installer.rb:28:in `install'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/source/rubygems.rb:203:in `install'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/installer/gem_installer.rb:54:in `install'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/installer/parallel_installer.rb:130:in `do_install'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/installer/parallel_installer.rb:121:in `block in worker_pool'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/worker.rb:62:in `apply_func'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/worker.rb:57:in `block in process_queue'
  <internal:kernel>:187:in `loop'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/worker.rb:54:in `process_queue'
  ~/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.4.21/lib/bundler/worker.rb:90:in `block (2 levels) in create_threads'

An error occurred while installing hpricot (0.8.6), and Bundler cannot continue.

In Gemfile:
  html2slim-ruby3 was resolved to 0.2.1, which depends on
    hpricot
```

### Using 

[html2slim-ruby3](https://rubygems.org/gems/html2slim-ruby3)

ruby -v
ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [arm64-darwin23]

rails -v
Rails 7.0.8.1

**Comment by tdegrunt** [comment]
Luckily this is solvable, look at the output and not the warning type:
```
fast_xs.c:165:39: error: incompatible function pointer types passing 'VALUE (VALUE)' (aka 'unsigned long (unsigned long)') to parameter of type 'VALUE (*)(VALUE, VALUE)' (aka 'unsigned long (*)(unsigned long,
unsigned long)') [-Wincompatible-function-pointer-types]
```

Then when installing the gem pass the --with-cflags option:
```
gem install html2slim -- --with-cflags="-Wno-incompatible-function-pointer-types"
```

This should allow you to install html2slim.

**Comment by viktorianer** [comment]
@tdegrunt Thanks a lot. Setting the following flags in `.bundle/config` helped 👍🏼 

```
# .bundle/config
BUNDLE_BUILD__HPRICOT: "--with-cflags=-Wno-incompatible-function-pointer-types"
```

But does not solve the issue itself, right? I would like to keep it open, for the reference.
Not sure, if someone ever will fix it.

**Comment by minad** [comment]
Fixed by #48

**Comment by viktorianer** [comment]
Superb! Thank you, @sanadan!
